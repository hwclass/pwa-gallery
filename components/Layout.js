import Head from 'next/head'

import Header from '../components/Header'

export default class extends React.Component {
  render() {
    return (
      <div>
        <Head>
          <title>{this.props.title}</title>
          <meta name="viewport" content="width=device-width, initial-scale=1.0"/>
          <link rel="manifest" href="/static/manifest.webmanifest" />
          <meta name="theme-color" content="#ff6600" />
          <link rel="apple-touch-icon" href="/static/icon.png" />
          <meta name="apple-mobile-web-app-title" content="PWA Gallery" />
          <meta name="apple-mobile-web-app-status-bar-style" content="default" />
          <meta name="apple-mobile-web-app-capable" content="yes" />
          <meta name="mobile-web-app-capable" content="yes" />
        </Head>
        <Header text={this.props.headerText}/>
        {this.props.children}
        <style jsx global>{`
          body {
            background: #fff;
            font: 11px Helvetica;
            font-size: 18px;
            padding: 0;
            margin: 0;
          }
        `}</style>
        <style jsx>{`
          display: grid;
          grid-template-columns: repeat(1, 1fr);
          grid-auto-rows: minmax(50px, auto);
          grid-template-areas:
            'hd'
            'main'
            'ft';
        `}</style>
      </div>
    );
  }
}
