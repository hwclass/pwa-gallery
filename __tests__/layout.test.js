/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Layout from '../components/Layout'

describe('Layout', () => {
  it('should render', () => {
    const component = shallow(<Layout title='Title' headerText='Header Text'/>)
    expect(component.getElements()).toMatchSnapshot();
  })
})

describe('Layout', () => {
  it('should match with snapshot', () => {
    const component = renderer.create(<Layout title='Title' headerText='Header Text'/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})