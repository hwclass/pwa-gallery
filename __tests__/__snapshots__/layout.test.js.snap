// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match with snapshot 1`] = `
<div
  className="jsx-3488773145"
>
  <header
    className="jsx-1922746274"
  >
    Header Text
  </header>
</div>
`;

exports[`Layout should render 1`] = `
Array [
  <div
    className="jsx-3488773145"
  >
    <SideEffect(Head)>
      <title
        className="jsx-3488773145"
      >
        Title
      </title>
      <meta
        className="jsx-3488773145"
        content="width=device-width, initial-scale=1.0"
        name="viewport"
      />
      <link
        className="jsx-3488773145"
        href="/static/manifest.webmanifest"
        rel="manifest"
      />
      <meta
        className="jsx-3488773145"
        content="#ff6600"
        name="theme-color"
      />
      <link
        className="jsx-3488773145"
        href="/static/icon.png"
        rel="apple-touch-icon"
      />
      <meta
        className="jsx-3488773145"
        content="PWA Gallery"
        name="apple-mobile-web-app-title"
      />
      <meta
        className="jsx-3488773145"
        content="default"
        name="apple-mobile-web-app-status-bar-style"
      />
      <meta
        className="jsx-3488773145"
        content="yes"
        name="apple-mobile-web-app-capable"
      />
      <meta
        className="jsx-3488773145"
        content="yes"
        name="mobile-web-app-capable"
      />
    </SideEffect(Head)>
    <_class
      text="Header Text"
    />
    <JSXStyle
      css="body{background:#fff;font:11px Helvetica;font-size:18px;padding:0;margin:0;}"
      styleId="2456803378"
    />
    <JSXStyle
      css=".jsx-3488773145{display:grid;grid-template-columns:repeat(1,1fr);grid-auto-rows:minmax(50px,auto);grid-template-areas: 'hd' 'main' 'ft';}"
      styleId="3091863714"
    />
  </div>,
]
`;
